packages/scene-composer/src/components/panels/scene-settings/MotionIndicatorVisibilityToggle.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useContext } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Toggle } from '@awsui/components-react';
+
+import { useStore, useViewOptionState } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { KnownComponentType } from '../../../interfaces';
+
+export const MotionIndicatorVisibilityToggle: React.FC = () => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const { motionIndicatorVisible, toggleMotionIndicatorVisibility } = useViewOptionState(sceneComposerId);
+  const getComponentRefByType = useStore(sceneComposerId)((state) => state.getComponentRefByType);
+  const { formatMessage } = useIntl();
+
+  const hasMotionIndicator = Object.keys(getComponentRefByType(KnownComponentType.MotionIndicator)).length > 0;
```

Comment:
You can use useMemo to memorize the value.

Reply:
Ultimately we can use `userpreference` settings to rememeber the state of the toggle button.

Reply:
sure